libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/SkillMocks/SkillClaimMock.cs

```diff
@@ -0,0 +1,88 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Connector.Authentication;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.SkillMocks
+{
+    public class SkillClaimMock : SkillMock
+    {
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.Test.SkillClaimMock";
+
+        [JsonConstructor]
+        public SkillClaimMock([CallerFilePath] string path = "", [CallerLineNumber] int line = 0)
+        {
+            RegisterSourcePath(path, line);
+        }
+
+        /// <summary>
+        /// Gets or sets the version claim.
+        /// </summary>
+        /// <value>
+        /// The version claim. Default is 2.0.
+        /// </value>
+        [DefaultValue("2.0")]
+        [JsonProperty("versionClaim")]
+        public string VersionClaim { get; set; } = "2.0";
+
+        /// <summary>
+        /// Gets or sets the audience claim.
+        /// </summary>
+        /// <value>
+        /// The audience claim.
+        /// </value>
+        [JsonProperty("audienceClaim")]
+        public string AudienceClaim { get; set; }
+
+        /// <summary>
+        /// Gets or sets the authorized party.
+        /// </summary>
+        /// <value>
+        /// The authorized party.
+        /// </value>
+        [JsonProperty("authorizedParty")]
+        public string AuthorizedParty { get; set; }
+
+        public override void Setup(TestAdapter adapter)
+        {
+            if (VersionClaim == null || VersionClaim != "2.0")
+            {
+                throw new Exception("Version must be 2.0!");
+            }
+
+            var claims = new List<Claim>();
+            claims.Add(new Claim(AuthenticationConstants.VersionClaim, VersionClaim));
+            claims.Add(new Claim(AuthenticationConstants.AudienceClaim, string.IsNullOrEmpty(AudienceClaim) ? Guid.NewGuid().ToString() : AudienceClaim));
+            claims.Add(new Claim(AuthenticationConstants.AuthorizedParty, string.IsNullOrEmpty(AuthorizedParty) ? Guid.NewGuid().ToString() : AuthorizedParty));
+            var claimsIdentity = new ClaimsIdentity(claims);
+
+            adapter.Use(new MockSkillClaimMiddleware(claimsIdentity));
+        }
+
+        private class MockSkillClaimMiddleware : IMiddleware
+        {
+            private readonly ClaimsIdentity claimsIdentity;
+
+            public MockSkillClaimMiddleware(ClaimsIdentity claimsIdentity)
+            {
+                this.claimsIdentity = claimsIdentity;
+            }
+
+            public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default)
+            {
+                turnContext.TurnState.Add(BotAdapter.BotIdentityKey, claimsIdentity);
```

Comment:
>BotIdentityKey [](start = 53, length = 14)

Yeah, this is not a SkillMock...this is a ClaimsMock.

Reply:
@tomlm Yes, I just follow the code here https://github.com/microsoft/botbuilder-dotnet/blob/53a5648947ad4a46afdc164aae6bbec4b1f3cc6a/tests/Microsoft.Bot.Builder.Dialogs.Tests/DialogExtensionsTests.cs#L143 to make dialog work like being called as a skill (for example, test whether it will return correct information in EoC event).

**It is not a test for two dialogs that one is host and one is skill.**

Or we just use a bool in TestScript.cs? To make it simple since all these fields in SkillClaimMock are useless.

What's your opinion for this test scenario? Thanks!

Reply:
Now I use a boolean `CalledAsLeafSkill` instead to make it simple and clear.